### PR TITLE
Enable compressed payloads on ThreadStats object

### PR DESCRIPTION
**Pull request by csssuf**
### What does this PR do?

Enable submitting compressed payloads to the Datadog API.

### Motivation

We use lambdas to scrape metrics which we can't easily collect by other means. These metric lambdas can collect large numbers of metrics quickly, and occasionally submit too many metrics within the same flush interval for the Datadog API to handle. This is similar to the issue in DataDog/datadogpy#465. The solution implemented to fix that issue was compression of payloads submitted to the DD API.

### Testing Guidelines

`./scripts/run_tests.sh`

### Additional Notes

n/a

### Types of Changes

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

**Comment by tianchu** [comment]
Hi @csssuf, I'm closing the PR due to the lack of activity here. Feel free to reopen if this problem still persists and you would like to move forward. I also want to mention a new way (using the Lambda extension) to submit metrics directly from your Lambda functions (without using the forwarder)  https://docs.datadoghq.com/serverless/datadog_lambda_library/extension/.

**Comment by jeid64** [comment]
Please reopen this issue, it has a lack of activity as the datadog maintainers didn't respond. @tianchu the PR is ready to merge.

**Comment by tianchu** [comment]
Thanks for the confirmation that this PR is still relevant! we will prioritize some testing before merging. Sorry for the long wait 🙇

**Comment by tianchu** [comment]
I could successfully reproduce the issue by submitting tons of data points within 10s, and enabling `compress_payload` did fix the issue. I'm going to merge this PR and release in the next version. Thanks for your contribution!

However, I do want to point out, although `compress_payload` does help workaround the issue, submitting a really large payload does add a noticeable overhead to the function duration. Not sure if this is the case, but if you are looping through a huge list and submitting a data point for each item, it would be way more efficient if you aggregate those data points in the memory and then submit one aggregated value at the end of the loop. 

```python
# inefficient
for i in range(1000000):
    lambda_metric(
        metric_name='my.metric',
        value=i+1,
        tags=['tag1:value1']
    )
    
# efficient
sum = 0
for i in range(1000000):
    sum = sum + i + 1
lambda_metric(
    metric_name='my.metric',
    value=sum,
    tags=['tag1:value1']
)
 ```

**Review by tianchu — APPROVED** [review]

**Pull request merged**
